crypto/objects/o_names.c

```diff
@@ -66,8 +66,14 @@ static int obj_name_cmp(const OBJ_NAME *a, const OBJ_NAME *b);
 static CRYPTO_ONCE init = CRYPTO_ONCE_STATIC_INIT;
 DEFINE_RUN_ONCE_STATIC(o_names_init)
 {
-    names_lh = lh_OBJ_NAME_new(obj_name_hash, obj_name_cmp);
+    names_lh = NULL;
     obj_lock = CRYPTO_THREAD_lock_new();
+    if (obj_lock != NULL)
```

Comment:
Might it be more readable to have:
```c
names_lh = ...;
obj_lock = ...;
if (names_lh != NULL && obj_lock != NULL)
    return 1;
free(names_lh);
free(obj_lock);
return 0;
```

Reply:
I pondered over this, but correctly freeing names_lh is rather complicated,
while freeing obj_lock is very simple.